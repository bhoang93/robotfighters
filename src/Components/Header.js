import React, {Component} from 'react';
import CounterButton from './CounterButton';

	class Header extends Component {
		render() {
			return (<div>
			<h1 className='f1'>Super Robot Fighting League</h1>
				<CounterButton colour="red"/>
			</div>)
		}
	}

export default Header;