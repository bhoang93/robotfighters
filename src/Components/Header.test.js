import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

it('expects to render Card component', () => {
	expect(shallow(<Header CounterButton={<button>Counter</button>}/>)).toMatchSnapshot();
})
