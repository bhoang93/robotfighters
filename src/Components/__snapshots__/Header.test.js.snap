// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render Card component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    CounterButton={
        <button>
            Counter
        </button>
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="f1"
>
          Super Robot Fighting League
</h1>,
        <CounterButton
          colour="red"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Super Robot Fighting League",
          "className": "f1",
        },
        "ref": null,
        "rendered": "Super Robot Fighting League",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "colour": "red",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="f1"
>
            Super Robot Fighting League
</h1>,
          <CounterButton
            colour="red"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Super Robot Fighting League",
            "className": "f1",
          },
          "ref": null,
          "rendered": "Super Robot Fighting League",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "colour": "red",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
